libp2p/protocols/secure/noise.nim

```diff
@@ -505,11 +505,15 @@ method handshake*(p: Noise, conn: Connection, initiator: bool): Future[SecureCon
           received_key = $remotePubKey
         raise newException(NoiseHandshakeError, "Noise handshake, peer infos don't match! " & $pid & " != " & $conn.peerInfo.peerId)
 
-    let peerInfo =
-      if conn.peerInfo != nil: conn.peerInfo
-      else: PeerInfo.init(remotePubKey)
+    let peerId = PeerID.init(remotePubKey).tryGet()
```

Comment:
why create peerid if it's only used in else branch below?

Reply:
I had a check to see if peerids matched where this was used, which I just removed - leftover.